drivers/video/v4l2driver.cpp

```diff
@@ -806,6 +854,16 @@ bool V4L2_Driver::ISNewNumber(const char * dev, const char * name, double values
         return true;
     }
 
+    // Ensure duration is set to one of the allowed durations, or, if we're going to stack, a value greater than the maximum
+    if (isIOptron() && strcmp(name, "CCD_EXPOSURE") == 0 && values[0] < MAX_IOPTRON_DURATION) {
+        int n = findIndexOfNearestValue(values[0], IOptronDurations);
+        if (values[0] != IOptronDurations[n]) 
+        {
+            LOGF_WARN("Changing duration to nearest value supported by camera (from %fs to %fs)", values[0], IOptronDurations[n]);
```

Comment:
Could this have the effect of being spammy if a user keep using the same non-standard exposure repeatedly only to this this warning over and over?